### PR TITLE
BOAC-1793: Adds category, topics, attachments to note feed

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1793

### CODE REVIEW

File: boac/merged/student.py [hunk 1]
```diff
@@ -399,14 +399,20 @@ def _note_to_json(note):
...
 def get_advising_notes(sid):
     notes_result = data_loch.get_advising_notes(sid)
     if not notes_result:
         return None
-    notes_by_id = {row['id']: _note_to_json(row) for row in notes_result}
+    notes_by_id = {}
+    for row in notes_result:
+        row['topics'] = data_loch.get_advising_note_topics(row['id'])
+        row['attachments'] = data_loch.get_advising_note_attachments(row['id'])
+        notes_by_id[row['id']] = _note_to_json(row)
```
Comment (johncrossman) [inline boac/merged/student.py]: @lyttam If a student has a dozen notes (not uncommon) then the loop above adds 24 more queries to the loch. Would a bulk query be possible?

File: fixtures/loch.sql [hunk 2]
```diff
@@ -61,17 +61,38 @@ CREATE TABLE boac_advising_notes.advising_notes
...
     created_at TIMESTAMP WITH TIME ZONE NOT NULL,
     updated_at TIMESTAMP WITH TIME ZONE NOT NULL
 );
 
+CREATE TABLE boac_advising_notes.advising_note_topics
+(
+    advising_note_id VARCHAR NOT NULL,
+    sid VARCHAR NOT NULL,
+    student_note_nr VARCHAR NOT NULL,
+    note_topic VARCHAR NOT NULL
```
Comment (johncrossman) [inline fixtures/loch.sql]: @lyttam do we need `sid` and `student_note_nr` in this view?

File: fixtures/loch.sql [hunk 3]
```diff
@@ -61,17 +61,38 @@ CREATE TABLE boac_advising_notes.advising_notes
...
+(
+    advising_note_id VARCHAR NOT NULL,
+    sid VARCHAR NOT NULL,
+    student_note_nr VARCHAR NOT NULL,
+    attachment_seq_nr INT,
+    attachment_date DATE,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL,
+    sis_file_name VARCHAR NOT NULL
+);
```
Comment (johncrossman) [inline fixtures/loch.sql]: @lyttam same question here. We don't want to ever the JOIN on `sid`, right?

File: boac/merged/student.py (hunk 1 above)
Comment (lyttam) [reply boac/merged/student.py]: Yes, good idea... some students have around 30 notes.

File: fixtures/loch.sql (hunk 2 above)
Comment (lyttam) [reply fixtures/loch.sql]: No, I'll remove them.

**Comment by lyttam** [comment]
@johncrossman I took your suggestions, but left `sid` on the topics and attachments tables since we now need it to do the bulk queries.

**Comment by johncrossman** [comment]
Thanks @lyttam  Makes sense.

**Pull request merged**